static/app/views/lastKnownRouteContextProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import {createContext, useContext} from 'react';
+
+import getRouteStringFromRoutes from 'sentry/utils/getRouteStringFromRoutes';
+import usePrevious from 'sentry/utils/usePrevious';
+import {useRoutes} from 'sentry/utils/useRoutes';
+
+interface Props {
+  children?: React.ReactNode;
+}
+
+export const LastKnownRouteContext = createContext<string>('');
+
+export function useLastKnownRoute() {
+  return useContext(LastKnownRouteContext);
```

Comment:
Do we care if this hook ever gets called outside of the provider?

Reply:
hmm, i guess it _could_ possibly be useful elsewhere in the future, so it's not awful if it does get called outside of here. the only place we currently need it is inside `routeNotFound.tsx`